tests/nn/data_parallel/test_sharded_ddp_features.py

```diff
@@ -248,6 +255,29 @@ def test_random_attributes():
     dist.destroy_process_group()
 
 
+def test_catch_grad_grad():
+    # Check that ShardedDDP exposes the original module's attributes
+    dist.init_process_group(init_method="file://" + tempfile.mkstemp()[1], backend="gloo", rank=0, world_size=1)
```

Comment:
nit: using mkstemp and not deleting the file will leave it behind in /tmp. I have a tempfile context function in testing.py that you could use to ensure the temp file is removed. right now, our tests leave behind a lot of temp files in /tmp. :-(

Reply:
we could move all the tests like this indeed, would be nicer indeed, I'll add that to the backlog (it would be a "big" PR in terms of volume)